Jenkinsfile

```diff
@@ -183,12 +183,48 @@ node('ca-jenkins-agent') {
     // Check Vulnerabilities
     pipeline.checkVulnerabilities()
 
+    if (env.CHANGE_BRANCH) {
+        pipeline.createStage(
+            name: "Changelog Verification",
+            stage: {
+                sh "git --no-pager fetch"
+                def changedFiles = sh(returnStdout: true, script: "git --no-pager diff origin/master --name-only").trim()
+                if (changedFiles.contains("CHANGELOG.md")) {
+                    echo "Changelog has been modified from origin/master."
+                } else {
+                    error "Changelog has not been modified from origin/master."
+                }
+            }
+        )
+    }
+
     // Deploys the application if on a protected branch. Give the version input
     // 30 minutes before an auto timeout approve.
     pipeline.deploy(
         versionArguments: [timeout: [time: 30, unit: 'MINUTES']]
     )
 
+    pipeline.createStage(
+        name: "Update Changelog Version",
+        stage: {
+            def firstTen = sh(returnStdout: true, script: "head -10 CHANGELOG.md")
+            if (firstTen.contains("Recent Changes")) {
```

Comment:
Should this check be done earlier in the "Changelog Verification" stage? If there are any conditions that could result in an error, it would seem ideal to check for them before the deploy stage runs.